amqp/src/main/java/io/cloudevents/amqp/ProtonAmqpMessageFactory.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2018-Present The CloudEvents Authors
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.cloudevents.amqp;
+
+
+import javax.annotation.ParametersAreNonnullByDefault;
+
+import org.apache.qpid.proton.amqp.messaging.ApplicationProperties;
+import org.apache.qpid.proton.message.Message;
+
+import io.cloudevents.amqp.impl.ProtonAmqpBinaryMessageReader;
+import io.cloudevents.amqp.impl.ProtonAmqpMessageWriter;
+import io.cloudevents.amqp.impl.AmqpConstants;
+import io.cloudevents.core.message.MessageReader;
+import io.cloudevents.core.message.MessageWriter;
+import io.cloudevents.core.message.impl.GenericStructuredMessageReader;
+import io.cloudevents.core.message.impl.MessageUtils;
+import io.cloudevents.core.message.impl.UnknownEncodingMessageReader;
+import io.cloudevents.rw.CloudEventWriter;
+/**
+ * A factory class providing convenience methods for creating MessageReader and MessageWriter instances based on Qpid Proton.
+ */
+@ParametersAreNonnullByDefault
+public final class ProtonAmqpMessageFactory {
+
+    private ProtonAmqpMessageFactory() {
+        // prevent instantiation
+    }
+
+    /**
+     * Creates a MessageReader to read a proton-based {@link Message}.
+     * <p>
+     * This implementation simply calls {@link #createReader(String, ApplicationProperties, byte[])}.
+     * 
+     * @param message The proton message to read from.
+     * 
+     * @return        A message reader that can read the given proton message to a cloud event representation.
+     */
+    public static MessageReader createReader(final Message message) {
+
+        final byte[] payload = AmqpConstants.getPayloadAsByteArray(message.getBody());
+        return createReader(message.getContentType(), message.getApplicationProperties(), payload);
+    }
+
+    /**
+     * Creates a MessageReader using the content-type property and payload of a proton-based message.
+     * <p>
+     * This method simply calls {@link #createReader(String, ApplicationProperties, byte[])}.
+     * 
+     * @param contentType The content-type of the message payload.
+     * @param payload     The message payload in bytes.
+     * @return            A message reader capable of representing a CloudEvent from
+     *                    a message <em>content-type</em> property and <em>application-data</em>.
+     */
+    public static MessageReader createReader(final String contentType, final byte[] payload) {
```

Comment:
Hi @Alfusainey I just noticed this method and i was wondering... do we really need it? Can you provide a use case for it?